presto-native-execution/presto_cpp/main/operators/BroadcastFactory.cpp

```diff
@@ -137,9 +140,25 @@ BroadcastFileReader::BroadcastFileReader(
     velox::memory::MemoryPool* pool)
     : broadcastFileInfo_(std::move(broadcastFileInfo)),
       fileSystem_(fileSystem),
+      asyncDataCache_(velox::cache::AsyncDataCache::getInstance()),
       hasData_(true),
       numBytes_(0),
-      pool_(pool) {}
+      lease_(fileIds(), broadcastFileInfo_->filePath_),
+      pool_(pool) {
+  auto readFile = fileSystem_->openFileForRead(broadcastFileInfo_->filePath_);
+  numBytes_ = readFile->size();
+
+  bufferedInput_ = std::make_unique<velox::dwio::common::CachedBufferedInput>(
+      std::move(readFile),
+      velox::dwio::common::MetricsLog::voidLog(),
+      lease_.id(),
+      asyncDataCache_,
+      /*tracker=*/nullptr,
```

Comment:
What is tracker and groupId? Why is it Ok to hard-code these to null and 0 ?

Reply:
@vermapratyush  FYI, hive use directory name as group id
```
std::shared_ptr<FileHandle> FileHandleGenerator::operator()(
    const std::string& filename) {
  // We have seen cases where drivers are stuck when creating file handles.
  // Adding a trace here to spot this more easily in future.
  process::TraceContext trace("FileHandleGenerator::operator()");
  uint64_t elapsedTimeUs{0};
  std::shared_ptr<FileHandle> fileHandle;
  {
    MicrosecondTimer timer(&elapsedTimeUs);
    fileHandle = std::make_shared<FileHandle>();
    fileHandle->file = filesystems::getFileSystem(filename, properties_)
                           ->openFileForRead(filename);
    fileHandle->uuid = StringIdLease(fileIds(), filename);
    fileHandle->groupId = StringIdLease(fileIds(), groupName(filename));
    VLOG(1) << "Generating file handle for: " << filename
            << " uuid: " << fileHandle->uuid.id();
  }
  REPORT_ADD_HISTOGRAM_VALUE(
      kCounterHiveFileHandleGenerateLatencyMs, elapsedTimeUs / 1000);
  // TODO: build the hash map/etc per file type -- presumably after reading
  // the appropriate magic number from the file, or perhaps we include the file
  // type in the file handle key.
  return fileHandle;
}
```

Reply:
Updated this to use tracker from `Connector::getTracker` which will create a new tracker of not exist. The trackerId is same as scanId_.
At the moment scanId_ == broadcastFilePath, let me know if there is a better alternative to using broadcastFilePath.
In 
https://github.com/facebookincubator/velox/blob/918156e2e67093afda06cbf39e0af8295d3764d7/velox/connectors/hive/HiveConnector.cpp#L95
we use scanId from `connectorQueryCtx`, however connectorQueryCtx is not passed down to ExchangeSource.